src/main/java/de/hysky/skyblocker/skyblock/entity/MobGlow.java

```diff
@@ -2,81 +2,140 @@
+			cachedGlow = glowCache.get(entity);
+		}
 
-					// Regular Mobs
-					default -> SkyblockerConfigManager.get().dungeons.starredMobGlow && isStarred(entity);
-				};
-			}
+		return cachedGlow.value && playerCanSee(entity, currentTime);
+	}
 
-			return switch (entity) {
-				// Rift
-				case PlayerEntity p when Utils.isInTheRift() && !entity.isInvisible() && name.equals("Blobbercyst ") -> SkyblockerConfigManager.get().otherLocations.rift.blobbercystGlow;
 
-				// Enderman Slayer
-				// Highlights Nukekubi Heads
-				case ArmorStandEntity armorStand when Utils.isInTheEnd() && SlayerUtils.isInSlayer() && isNukekubiHead(armorStand) -> SkyblockerConfigManager.get().slayers.endermanSlayer.highlightNukekubiHeads;
+	/**
+	 * Checks if the player can see the entity.
+	 * Has "True sight" within a certain aura, but since name tags exist I think this is fine...
+	 */
+	private static boolean playerCanSee(Entity entity, long currentTime) {
 
-				// Special Zelot
-				case EndermanEntity enderman when Utils.isInTheEnd() && !entity.isInvisible() -> TheEnd.isSpecialZealot(enderman);
+		CacheEntry canSee = canSeeCache.get(entity);
+		if (canSee == null || (currentTime - canSee.timestamp) > PLAYER_CAN_SEE_CACHE_DURATION) {
+			boolean playerCanSee = entity.distanceTo(MinecraftClient.getInstance().player) <= 20 || MinecraftClient.getInstance().player.canSee(entity);
+			canSeeCache.put(entity, new CacheEntry(playerCanSee, currentTime));
+			return playerCanSee;
+		}
 
-				//dojo
-				case ZombieEntity zombie when Utils.isInCrimson() && DojoManager.inArena -> DojoManager.shouldGlow(getArmorStandName(zombie));
+		return canSee.value;
+	}
 
-				//Kuudra
-				case MagmaCubeEntity magmaCube when Utils.isInKuudra() -> SkyblockerConfigManager.get().crimsonIsle.kuudra.kuudraGlow && magmaCube.getSize() == Kuudra.KUUDRA_MAGMA_CUBE_SIZE;
+	private static boolean computeShouldMobGlow(Entity entity) {
+		String name = entity.getName().getString();
 
-				default -> false;
+		// Dungeons
+		if (Utils.isInDungeons()) {
+			return switch (entity) {
+				// Minibosses
+				case PlayerEntity p when name.equals("Lost Adventurer") || name.equals("Shadow Assassin") || name.equals("Diamond Guy") ->
+						SkyblockerConfigManager.get().dungeons.starredMobGlow;
+				case PlayerEntity p when entity.getId() == LividColor.getCorrectLividId() ->
+						LividColor.shouldGlow(name);
+				// Bats
+				case BatEntity b ->
+						SkyblockerConfigManager.get().dungeons.starredMobGlow || SkyblockerConfigManager.get().dungeons.starredMobBoundingBoxes;
+
+				// Armor Stands
+				case ArmorStandEntity _armorStand -> false;
+
+				// Regular Mobs
+				default -> SkyblockerConfigManager.get().dungeons.starredMobGlow && isStarred(entity);
 			};
 		}
 
-		return false;
+		return switch (entity) {
+
+			// Rift Blobbercyst
+			case PlayerEntity p when Utils.isInTheRift() && name.equals("Blobbercyst ") ->
+					SkyblockerConfigManager.get().otherLocations.rift.blobbercystGlow;
```

Comment:
There's a lot of unnecessary wrapping like this one in this file. Can you set your `Editor -> Code Style -> Java -> Wrapping and Braces -> Hard wrap at:` to 1000 and then reformat this.